src/main/java/com/alvarium/sign/Ed25519Provider.java

```diff
@@ -0,0 +1,40 @@
+package com.alvarium.sign;
+
+import java.security.GeneralSecurityException;
+
+import com.alvarium.utils.Encoder;
+import com.google.crypto.tink.subtle.Ed25519Sign;
+import com.google.crypto.tink.subtle.Ed25519Verify;
+
+public class Ed25519Provider implements SignProvider {
```

Comment:
Please provide a `protected` constructor to avoid instantiation by classes outside of this package. It might make sense to move `byte[] key` from the `sign()` method to a constructor parameter. Consider that in light of how/whether it negatively impacts the extensibility of the factory. I'm pretty sure all signature implementations will require a key provided as `[]byte` but worth a look.

Reply:
Thanks for pointing out the first issue, I'll push that change as well.

@KarimElghamry and I actually briefly discussed passing `byte[] key` through the constructor instead and move all instantiation logic there, but didn't want to mess with the interface, since that's a possibility though I'll definitely look deeper into it.